src/rotator_library/providers/utilities/player2_auth.py

```
@@ -0,0 +1,254 @@
+    async with httpx.AsyncClient(timeout=15.0, base_url=PLAYER2_API_BASE) as client:
+        new_resp = await client.post("/login/device/new", json={"client_id": client_id})
+        new_resp.raise_for_status()
+        data = new_resp.json()
+
+        device_code = data["deviceCode"]
+        user_code = data["userCode"]
+        verification_uri = data.get("verificationUriComplete") or data.get("verificationUri")
+        if not verification_uri:
+            raise Player2AuthError(
+                "Player2 did not return a verification URI for the device login."
+            )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)